From 2d98385665890514bc74b7a365c968a6bca335c5 Mon Sep 17 00:00:00 2001
From: Nizan Zorea <nzorea@marvell.com>
Date: Mon, 8 Feb 2016 18:54:58 +0200
Subject: [PATCH 0922/1240] fdt_config: mvebu: add support to print which FDT
 is active

- add the option to print FDT active selection with "fdt_config select".

Change-Id: Ib0094df4de24af7937125d5522b28b984958802a
Signed-off-by: Nizan Zorea <nzorea@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27369
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 board/mvebu/common/fdt_config.c | 21 +++++++++++++++++++++
 board/mvebu/common/fdt_config.h |  1 +
 common/mvebu/cmd_fdt_config.c   | 15 ++++++++-------
 3 files changed, 30 insertions(+), 7 deletions(-)

diff --git a/board/mvebu/common/fdt_config.c b/board/mvebu/common/fdt_config.c
index 508276b..38bee1d 100644
--- a/board/mvebu/common/fdt_config.c
+++ b/board/mvebu/common/fdt_config.c
@@ -54,6 +54,26 @@ int fdt_create_list(void)
 	return 0;
 }
 
+/* fdt_select_print - print active FDT selection */
+void fdt_select_print(void)
+{
+	int i;
+	struct eeprom_struct *p_board_config;
+
+	if (fdt_list_size == -1)
+		fdt_create_list();
+
+	p_board_config = cfg_eeprom_get_board_config();
+	for (i = 0; i < fdt_list_size; i++) {
+		if (p_board_config->board_config.active_fdt_selection == fdt_list_of_configs[i].fdt_config_id) {
+			printf("\t%d - %s\n", fdt_list_of_configs[i].fdt_config_id, fdt_list_of_configs[i].fdt_model);
+			return;
+		}
+	}
+
+	return;
+}
+
 /* fdt_select_set - update active_fdt_selection field */
 int fdt_select_set(const char *selected_index)
 {
@@ -89,6 +109,7 @@ int fdt_select_list(void)
 	printf("FDT config list:\n");
 	for (i = 0; i < fdt_list_size; i++)
 		printf("\t%d - %s\n", fdt_list_of_configs[i].fdt_config_id, fdt_list_of_configs[i].fdt_model);
+
 	return 0;
 }
 
diff --git a/board/mvebu/common/fdt_config.h b/board/mvebu/common/fdt_config.h
index a5c4179..2a83e15 100644
--- a/board/mvebu/common/fdt_config.h
+++ b/board/mvebu/common/fdt_config.h
@@ -25,5 +25,6 @@ int fdt_cfg_on(void);
 int fdt_cfg_off(void);
 int fdt_select_set(const char *selected_index);
 int fdt_select_list(void);
+void fdt_select_print(void);
 
 #endif
diff --git a/common/mvebu/cmd_fdt_config.c b/common/mvebu/cmd_fdt_config.c
index ff99039..d8c6af7 100644
--- a/common/mvebu/cmd_fdt_config.c
+++ b/common/mvebu/cmd_fdt_config.c
@@ -31,9 +31,6 @@ int do_fdt_config_cmd(cmd_tbl_t *cmdtp, int flag, int argc,
 	const char *fdt_option = NULL;
 	const char *fdt_model = NULL;
 
-	if ((strcmp(cmd, "select") == 0) && (argc < 3))
-		return CMD_RET_USAGE;
-
 	if (argc > 2)
 		fdt_option = argv[2];
 
@@ -57,8 +54,11 @@ int do_fdt_config_cmd(cmd_tbl_t *cmdtp, int flag, int argc,
 		if (fdt_cfg_off())
 			return 1;
 	} else if (strcmp(cmd, "select") == 0) {
-		if (fdt_select_set(fdt_option))
-			return 1;
+		if (argc < 3) {
+			fdt_select_print();
+		} else if (fdt_select_set(fdt_option)) {
+				return 1;
+			}
 	} else if (strcmp(cmd, "list") == 0) {
 		if (fdt_select_list())
 			return 1;
@@ -71,14 +71,15 @@ int do_fdt_config_cmd(cmd_tbl_t *cmdtp, int flag, int argc,
 
 U_BOOT_CMD(
 	fdt_config,    6,     1,      do_fdt_config_cmd,
-	"fdt_config - Modify SOC and board configuration\n",
+	"Modify SOC and board configuration\n",
 	"\n"
 	"Modify SOC and board configuration\n"
-	"\tread	eeprom	  - Read FDT from EEPROM and save to DRAM\n"
+	"\tread eeprom	  - Read FDT from EEPROM and save to DRAM\n"
 	"\tread flash <x> - Read x FDT from U-Boot and save to DRAM\n"
 	"\tsave		  - Save FDT in EEPROM\n"
 	"\toff		  - Disable the feature of loading the FDT that saved in EEPROM\n"
 	"\ton		  - Enable the feature of loading the FDT that saved in EEPROM\n"
 	"\tlist		  - Show the options of the board\n"
+	"\tselect	  - Print active FDT selection\n"
 	"\tselect <x>	  - Update active FDT selection\n"
 );
-- 
1.9.1

